src/main/resources/assets/computercraft/lua/rom/programs/monitor.lua

```diff
@@ -39,7 +39,40 @@ local function resume( ... )
     return param
 end
 
-local ok, param = pcall( function()
+local timers = { n = 0 }
+
+-- add a timer to the timers list with id 'id', xpos 'x', ypos 'y'
+local function addTimer( id, x, y )
+    timers.n = timers.n + 1
+    timers[timers.n] = { n = 3, id, x, y }
+end
+
+-- remove a timer from the timers list with id 'id'
+local function removeTimer( id )
+    for i = 1, timers.n do
+        if timers[i][1] == id then
+            local x, y = table.unpack( timers[i], 2, timers[i].n )
+            table.remove( timers, i )
+            timers.n = timers.n - 1
+            return true, x, y
+        end
+    end
+    return false
+end
+
+local ok, param = pcall( parallel.waitForAny,
```

Comment:
Why are you using parallel instead on integrating this logic into main event loop?